opencti-platform/opencti-graphql/src/database/engine.js

```diff
@@ -1232,41 +1231,24 @@ export const elFindByIds = async (context, user, ids, opts = {}) => {
     }
     // With onRelationship option, the ids needs to be found on a relationship connection side.
     // onRelationship must be from or to
-    if (onRelationship && (onRelationship === 'from' || onRelationship === 'to')) {
-      const nested = {
-        nested: {
-          path: 'connections',
-          query: {
-            bool: {
-              must: [
-                { query_string: { query: `*_${onRelationship}`, fields: ['connections.role'] } },
-                { terms: { [`connections.${ID_INTERNAL}.keyword`]: workingIds } }
-              ]
-            },
-          },
-        }
-      };
-      mustTerms.push(nested);
-    } else {
-      // Compute combination terms
-      for (let indexType = 0; indexType < elementTypes.length; indexType += 1) {
-        const elementType = elementTypes[indexType];
-        const terms = { [`${elementType}.keyword`]: workingIds };
-        idsTermsPerType.push({ terms });
-      }
-      const should = {
-        bool: {
-          should: idsTermsPerType,
-          minimum_should_match: 1,
-        },
-      };
-      mustTerms.push(should);
+    // Compute combination terms
+    for (let indexType = 0; indexType < elementTypes.length; indexType += 1) {
+      const elementType = elementTypes[indexType];
+      const terms = { [`${elementType}.keyword`]: workingIds };
+      idsTermsPerType.push({ terms });
     }
+    const should = {
+      bool: {
+        should: idsTermsPerType,
+        minimum_should_match: 1,
+      },
+    };
+    mustTerms.push(should);
     if (types && types.length > 0) {
       const typesShould = types.map((typeShould) => (
         [
-          { match_phrase: { 'entity_type.keyword': typeShould } },
-          { match_phrase: { 'parent_types.keyword': typeShould } }
+          { term: { 'entity_type.keyword': typeShould } },
```

Comment:
here we could also do a terms query with the list of types instead of a list of should with one term.

Reply:
Updated